static-site/package.json

```diff
@@ -11,8 +11,8 @@
   "license": "MIT",
   "main": "n/a",
   "engines": {
-    "node": "^14",
-    "npm": "^6.14"
+    "node": "^16",
```

Comment:
Using this line in the code to get a 🧵, but the error isn't related to node-16 AFAIK.

One of the benefits of this was to allow native (arm64) installs of node, as eariler versions weren't available on conda for M1 architectures. However I can't seem to build the static-site under a M1 architecture as sharp (used in the static site via gatsby-sharp-plugin) doesn't seem to be useable on M1 - see https://github.com/lovell/sharp/issues/2533.

Initially I got the following stack trace:


<details>
  <summary>Stack trace of build failure related to gatsby-sharp-plugin</summary>
  
  ```
  bash build.sh static
  Building the static site (./static-site/public/)
  npm WARN deprecated @types/vfile-message@2.0.0: This is a stub types definition. vfile-message provides its own type definitions, so you do not need this installed.
  npm WARN deprecated ini@1.3.5: Please update to ini >=1.3.6 to avoid a prototype pollution issue
  npm WARN deprecated async-cache@1.1.0: No longer maintained. Use [lru-cache](http://npm.im/lru-cache) version 7.6 or higher, and provide an asynchronous `fetchMethod` option.
  npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
  npm WARN deprecated flatten@1.0.3: flatten is deprecated in favor of utility frameworks such as lodash.
  npm WARN deprecated @hapi/bourne@1.3.2: This version has been deprecated and is no longer supported or maintained
  npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
  npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
  npm WARN deprecated har-validator@5.1.3: this library is no longer supported
  npm WARN deprecated eslint-loader@2.2.1: This loader has been deprecated. Please use eslint-webpack-plugin
  npm WARN deprecated fsevents@2.1.3: “Please update to latest v2.3 or v2.2”
  npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
  npm WARN deprecated browserslist@1.7.7: Browserslist 2 could fail on reading Browserslist >3.0 config used in other tools.
  npm WARN deprecated browserslist@1.7.7: Browserslist 2 could fail on reading Browserslist >3.0 config used in other tools.
  npm WARN deprecated browserslist@1.7.7: Browserslist 2 could fail on reading Browserslist >3.0 config used in other tools.
  npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
  npm WARN deprecated querystring@0.2.1: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
  npm WARN deprecated chokidar@1.7.0: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
  npm WARN deprecated browserslist@2.11.3: Browserslist 2 could fail on reading Browserslist >3.0 config used in other tools.
  npm WARN deprecated source-map-url@0.4.0: See https://github.com/lydell/source-map-url#deprecated
  npm WARN deprecated circular-json@0.3.3: CircularJSON is in maintenance only, flatted is its successor.
  npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
  npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
  npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
  npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
  npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
  npm WARN deprecated @hapi/address@2.1.4: Moved to ‘npm install @sideway/address’
  npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
  npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
  npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
  npm WARN deprecated subscriptions-transport-ws@0.9.19: The `subscriptions-transport-ws` package is no longer maintained. We recommend you use `graphql-ws` instead. For help migrating Apollo software to `graphql-ws`, see https://www.apollographql.com/docs/apollo-server/data/subscriptions/#switching-from-subscriptions-transport-ws    For general help using `graphql-ws`, see https://github.com/enisdenjo/graphql-ws/blob/master/README.md
  npm WARN deprecated @hapi/joi@15.1.1: Switch to ‘npm install joi’
  npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
  npm WARN deprecated gatsby-recipes@0.9.3: gatsby-recipes has been removed from gatsby/gatsby-cli >=4.5.0. Update to gatsby@latest/gatsby-cli@latest to use versions without gatsby-recipes. This package will no longer receive updates.
  npm WARN deprecated core-js@1.2.7: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.
  npm WARN deprecated core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.
  npm ERR! code 1
  npm ERR! path /Users/enigma/projects/nextstrain/nextstrain.org/static-site/node_modules/sharp
  npm ERR! command failed
  npm ERR! command sh /var/folders/7w/rvnm5fgd7kb83fw8lg882x680000gn/T/install-190b74fb.sh
  npm ERR! CC(target) Release/obj.target/nothing/../node-addon-api/src/nothing.o
  npm ERR!   LIBTOOL-STATIC Release/nothing.a
  npm ERR!   TOUCH Release/obj.target/libvips-cpp.stamp
  npm ERR!   CXX(target) Release/obj.target/sharp/src/common.o
  npm ERR! info sharp Downloading https://github.com/lovell/sharp-libvips/releases/download/v8.9.1/libvips-8.9.1-darwin-arm64v8.tar.gz
  npm ERR! ERR! sharp Prebuilt libvips 8.9.1 binaries are not yet available for darwin-arm64v8
  npm ERR! info sharp Attempting to build from source via node-gyp but this may fail due to the above error
  npm ERR! info sharp Please see https://sharp.pixelplumbing.com/install for required dependencies
  npm ERR! gyp info it worked if it ends with ok
  npm ERR! gyp info using node-gyp@9.0.0
  npm ERR! gyp info using node@16.17.1 | darwin | arm64
  npm ERR! gyp info find Python using Python version 3.10.7 found at “/usr/local/opt/python@3.10/bin/python3.10”
  npm ERR! gyp info spawn /usr/local/opt/python@3.10/bin/python3.10
  npm ERR! gyp info spawn args [
  npm ERR! gyp info spawn args   ‘/Users/enigma/miniconda3/envs/node16/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py’,
  npm ERR! gyp info spawn args   ‘binding.gyp’,
  npm ERR! gyp info spawn args   ‘-f’,
  npm ERR! gyp info spawn args   ‘make’,
  npm ERR! gyp info spawn args   ‘-I’,
  npm ERR! gyp info spawn args   ‘/Users/enigma/projects/nextstrain/nextstrain.org/static-site/node_modules/sharp/build/config.gypi',
  npm ERR! gyp info spawn args   ‘-I’,
  npm ERR! gyp info spawn args   ‘/Users/enigma/miniconda3/envs/node16/lib/node_modules/npm/node_modules/node-gyp/addon.gypi’,
  npm ERR! gyp info spawn args   ‘-I’,
  npm ERR! gyp info spawn args   ‘/Users/enigma/Library/Caches/node-gyp/16.17.1/include/node/common.gypi’,
  npm ERR! gyp info spawn args   ‘-Dlibrary=shared_library’,
  npm ERR! gyp info spawn args   ‘-Dvisibility=default’,
  npm ERR! gyp info spawn args   ‘-Dnode_root_dir=/Users/enigma/Library/Caches/node-gyp/16.17.1’,
  npm ERR! gyp info spawn args   ‘-Dnode_gyp_dir=/Users/enigma/miniconda3/envs/node16/lib/node_modules/npm/node_modules/node-gyp’,
  npm ERR! gyp info spawn args   ‘-Dnode_lib_file=/Users/enigma/Library/Caches/node-gyp/16.17.1/<(target_arch)/node.lib’,
  npm ERR! gyp info spawn args   ‘-Dmodule_root_dir=/Users/enigma/projects/nextstrain/nextstrain.org/static-site/node_modules/sharp',
  npm ERR! gyp info spawn args   ‘-Dnode_engine=v8’,
  npm ERR! gyp info spawn args   ‘—depth=.’,
  npm ERR! gyp info spawn args   ‘—no-parallel’,
  npm ERR! gyp info spawn args   ‘—generator-output’,
  npm ERR! gyp info spawn args   ‘build’,
  npm ERR! gyp info spawn args   ‘-Goutput_dir=.’
  npm ERR! gyp info spawn args ]
  npm ERR! gyp info spawn make
  npm ERR! gyp info spawn args [ ‘BUILDTYPE=Release’, ‘-C’, ‘build’ ]
  npm ERR! warning: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: archive library: Release/nothing.a the table of contents is empty (no object file members in the library define global symbols)
  npm ERR! ../src/common.cc:23:10: fatal error: ‘vips/vips8’ file not found
  npm ERR! #include <vips/vips8>
  npm ERR!          ^~~~~~~~~~~~
  npm ERR! 1 error generated.
  npm ERR! make: *** [Release/obj.target/sharp/src/common.o] Error 1
  npm ERR! gyp ERR! build error 
  npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
  npm ERR! gyp ERR! stack     at ChildProcess.onExit (/Users/enigma/miniconda3/envs/node16/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
  npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
  npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)
  npm ERR! gyp ERR! System Darwin 20.6.0
  npm ERR! gyp ERR! command “/Users/enigma/miniconda3/envs/node16/bin/node” “/Users/enigma/miniconda3/envs/node16/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js” “rebuild”
  npm ERR! gyp ERR! cwd /Users/enigma/projects/nextstrain/nextstrain.org/static-site/node_modules/sharp
  npm ERR! gyp ERR! node -v v16.17.1
  npm ERR! gyp ERR! node-gyp -v v9.0.0
  npm ERR! gyp ERR! not ok

  npm ERR! A complete log of this run can be found in:
  npm ERR!     /Users/enigma/.npm/_logs/2022-11-30T23_30_18_685Z-debug-0.log

  Build failed at build.sh line 37

  You are responsible for clean up (sorry!)
  ```
</details>

I overcame this by running `brew install vips` (from [this stackoverflow answer](https://stackoverflow.com/questions/66363572/vips-vips8-file-not-found-include-vips-vips8-in-ionic-cordova-in-ios)). However now the static-site can't build:

```
  Error: Something went wrong installing the "sharp" module
  dlopen(/Users/enigma/projects/nextstrain/nextstrain.org/static-site/node_modules/sharp/build/Release/sharp.node, 1): Symbol not found: __ZTVN4vips7VOptionE
    Referenced from: /Users/enigma/projects/nextstrain/nextstrain.org/static-site/node_modules/sharp/build/Release/sharp.node
    Expected in: flat namespace
   in /Users/enigma/projects/nextstrain/nextstrain.org/static-site/node_modules/sharp/build/Release/sharp.node
  - Remove the "node_modules/sharp" directory then run
    "npm install --ignore-scripts=false --verbose" and look for errors
  - Consult the installation documentation at https://sharp.pixelplumbing.com/install
  - Search for this error at https://github.com/lovell/sharp/issues
```

I'll try creating a rosetta environment with node16 & see if that works...

Reply:
I also did `brew install vips` (which took forever!!!) but don't remember encountering issues afterwards. I can try again, though a better solution here might be to upgrade to a newer version of gatsby-plugin-sharp that uses sharp v0.28.0 (based on https://github.com/lovell/sharp/issues/2460#issuecomment-811046375).

Reply:
> I'll try creating a rosetta environment with node16 & see if that works...

Can confirm this works, and so the above is related to M1 (which won't be encountered by CI)

```
node -p "process.arch"
x64
```

Reply:
I presume `brew` is arch specific (it's gotta be...) and so I've probably got a osx-64 brew? This is not fun stuff & I don't like using `brew` because it's not scoped to the (conda) environment

Reply:
Actually I'm running into a different build error (https://github.com/webpack/webpack/issues/14532) since Webpack 4 is used in `static-site` (looks like that's the resolved Webpack version for the installed version of gatsby):

```
static-site % npm list webpack
static@1.0.1
├─┬ gatsby@2.32.13
│ ├─┬ @pieh/friendly-errors-webpack-plugin@1.7.0-chalk-2
│ │ └── webpack@4.46.0 deduped
│ ├─┬ @pmmmwh/react-refresh-webpack-plugin@0.4.3
│ │ └── webpack@4.46.0 deduped
│ ├─┬ babel-loader@8.2.5
│ │ └── webpack@4.46.0 deduped
│ ├─┬ css-loader@1.0.1
│ │ └── webpack@4.46.0 deduped
│ ├─┬ eslint-loader@2.2.1
│ │ └── webpack@4.46.0 deduped
│ ├─┬ file-loader@1.1.11
│ │ └── webpack@4.46.0 deduped
│ ├─┬ mini-css-extract-plugin@0.11.3
│ │ └── webpack@4.46.0 deduped
│ ├─┬ null-loader@3.0.0
│ │ └── webpack@4.46.0 deduped
│ ├─┬ optimize-css-assets-webpack-plugin@5.0.8
│ │ └── webpack@4.46.0 deduped
│ ├─┬ terser-webpack-plugin@2.3.8
│ │ └── webpack@4.46.0 deduped
│ ├─┬ url-loader@1.1.2
│ │ └── webpack@4.46.0 deduped
│ ├─┬ webpack-dev-middleware@3.7.3
│ │ └── webpack@4.46.0 deduped
│ ├─┬ webpack-dev-server@3.11.3
│ │ └── webpack@4.46.0 deduped
│ └─┬ webpack@4.46.0
│   └─┬ terser-webpack-plugin@1.4.5
│     └── webpack@4.46.0 deduped
└─┬ lodash-webpack-plugin@0.11.5
  └── webpack@4.46.0 deduped
```

In any case, like you mentioned, removing Gatsby would solve things.

Reply:
> I'll try creating a rosetta environment with node16 & see if that works...

I just ran into the same issue, but this didn't work for me 😕